### PR TITLE
Fix code to allow 0 in x-form-money input text

### PR DESCRIPTION
**Pull request by devloper-vijahat**
This is the fix to allow money input with 0.
issue describe here.

https://github.com/user-attachments/assets/725b9758-9db0-43b7-9a44-491dcb2b86f7

After solution it look like this

https://github.com/user-attachments/assets/d44a8e6d-1498-4ff4-9969-db9476327fbb

### CODE REVIEW

**Comment by BonnieSawe** [comment]
Hi @mikebronner check this PR out.

**Comment by mikebronner** [comment]
Hi @BonnieSawe @devloper-vijahat could you post the videos in a compatible format or maybe link to them where they are viewable? I am unable to view them.

**Comment by devloper-vijahat** [comment]
Hi @mikebronner here i am sharing link of the videos.
https://drive.google.com/file/d/1OC-OPeg1b7WM5hprbV93uusX1p0rpyZn/view
https://drive.google.com/file/d/1A4D4KWw4sVKHhGrH-U7CRE8UvOBMW233/view

**Comment by mikebronner** [comment]
@BonnieSawe @devloper-vijahat  I don't think the solution is quite right. Instead of the fix displaying just `0` it should display `0.00` and retain the format.

**Comment by devloper-vijahat** [comment]
Hey @mikebronner ,
Here i am sharing you a video after update the changes.
https://drive.google.com/file/d/1KHbhU0x4LIm3_I_Y6lijZvLHhzLkrYCU/view

Please review them and let me know if anything more require.
Thank you

File: resources/views/components/money.blade.php [hunk 1]
```diff
@@ -64,8 +64,12 @@ class="relative"
                         minimumFractionDigits: {{ $decimals }},
                         maximumFractionDigits: {{ $decimals }}
                     });
-                
-                if (displayValue == '0.00' || displayValue == 0) {
+
+                if (
+                    (displayValue == '0.00'
+                        || displayValue == 0)
+                    && this.value != 0
```
Comment (mikebronner) [inline resources/views/components/money.blade.php:71]: @devloper-vijahat  what is the value of `this.value` when display value is 0 or 0.00?

**Review by mikebronner — COMMENTED** [review]
looks good, just want to understand what is happening with this.value

File: resources/views/components/money.blade.php (hunk 1 above)
Comment (devloper-vijahat) [reply resources/views/components/money.blade.php:71]: @mikebronner  value of `this.value` when display value was 0 or 0.00 is the main value that we enter in text box like we add 0 so `this.value` was 0.

Comment (mikebronner) [reply resources/views/components/money.blade.php:71]: @devloper-vijahat @BonnieSawe  in that case this condition doesn't make sense, does it? when would this condition ever be true? Have you tested this with other random numbers other than 0 to see how it behaves?

Comment (devloper-vijahat) [reply resources/views/components/money.blade.php:71]: @mikebronner Yes i tested it with other number also it is working fine for other number also i will create a video and explain you about it.
https://drive.google.com/file/d/19Z-jcSF8SZ4Hv-sxjb5h2pmxXopdtUno/view

Here is the video link that show it working fine for random number.
In this we only check here if 0 enter by user then value was not disappear from input text that why we add this condition.

**Pull request merged**